jobs/travis.rb

```diff
@@ -4,131 +4,103 @@
-			if metrics['scrutinizer.quality']
-				quality = metrics['scrutinizer.quality'].round
-				item['items'] << {
-					'label' => 'Qual: ' + String(quality),
-					'class' => 'rating rating-quality rating-' + String(quality),
-					'url' => scrutinizer_link
-				}
-			end
-			if metrics['scrutinizer.test_coverage']
-				coverage = metrics['scrutinizer.test_coverage']
-				item['items'] << {
-					'label' => 'Covrg: ' + String((coverage*100).round) + '%',
-					'class' => 'rating rating-coverage rating-' + String((coverage*10).round),
-					'url' => scrutinizer_link
-				}
-			end
-			
-		end
-
-		item
-	end
-
-	# Sort by name, then by status
-	items.sort_by! do|item|
-		if item['class'] == 'bad'
-			[1,item['label']]
-		elsif item['class'] == 'good'
-			[2,item['label']]
-		else
-			[3,item['label']]
-		end
-	end
-	
-	if items != $lastTravisItems
-		send_event('travis', {
-			unordered: true,
-			items: items
-		})
-	end
-
-	$lastTravisItems = items
-
-end
\ No newline at end of file
+  travis_backend = TravisBackend.new
+  repo_slugs = []
+  builds = []
+
+  # Only look at release branches (x.y) and master, not at tags (x.y.z)
+  branch_whitelist = /^(\d+\.\d+$|master)/
+  branch_blacklist_by_repo = {}
+  branch_blacklist_by_repo = JSON.parse(ENV['TRAVIS_BRANCH_BLACKLIST']) if ENV['TRAVIS_BRANCH_BLACKLIST']
+
+  # TODO Move to configuration
+  repo_slug_replacements = [/(silverstripe-australia\/|silverstripe-labs\/|silverstripe\/|silverstripe-)/,'']
+
+  if ENV['ORGAS']
+    ENV['ORGAS'].split(',').each do |orga|
+      repo_slugs = repo_slugs.concat(travis_backend.get_repos_by_orga(orga).collect{|repo|repo['slug']})
+    end
+  end
+
+  if ENV['REPOS']
+    repo_slugs.concat(ENV['REPOS'].split(','))
+  end
+
+  repo_slugs.sort!
+
+  items = repo_slugs.map do |repo_slug|
+    label = repo_slug
+    label = repo_slug.gsub(repo_slug_replacements[0],repo_slug_replacements[1]) if repo_slug_replacements
+    item = {
+      'label' => label,
+      'class' => 'none',
+      'url' => '',
+      'items' => [],
+    }
+
+    # Travis info
+    repo_branches = travis_backend.get_branches_by_repo(repo_slug)
+
+    if repo_branches and repo_branches['branches'].length > 0
+      # Latest builds are listed under "branches", but their corresponding branch name
+      # is stored through the "commits" association
+      items = repo_branches['branches']
+        .select do |branch|
+          commit = repo_branches['commits'].find{|commit|commit['id'] == branch['commit_id']}
+          branch_name = commit['branch']
+
+          # Ignore branches not in whitelist
+          if not branch_whitelist.match(branch_name)
+            false
+          # Ignore branches specifically blacklisted
+          elsif branch_blacklist_by_repo.has_key?(repo_slug) and branch_blacklist_by_repo[repo_slug].include?(branch_name)
+            false
+          else
+            true
+          end
+        end
```

Comment:
Place the . on the previous line, together with the method call receiver.<br>Align the operands of an expression in an assignment spanning multiple lines.